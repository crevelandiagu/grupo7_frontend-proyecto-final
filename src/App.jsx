import { Typography, Button } from "@mui/material"
function App() {

  return (
    <>
      <Typography variant="h1">ABC Jobs</Typography>
      <Button variant="contained">Login</Button>
    </>
  )
}

export default App
