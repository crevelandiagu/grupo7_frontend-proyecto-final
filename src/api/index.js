export { default as candidateApi } from './candidateApi';
export { default as companyApi } from './companyApi';
export { default as companyEmployeesApi } from './companyEmployeesApi';
export { default as projectsApi } from './projectsApi';
export { default as searchToolApi } from './searchToolApi';
export { default as selectionProcess } from './selectionProcess';
export { default as performance } from './performanceApi';
