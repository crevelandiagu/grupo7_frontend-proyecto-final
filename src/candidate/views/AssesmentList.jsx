import { Grid, Typography } from '@mui/material';
import Box from '@mui/material/Box';
import { DataGrid } from '@mui/x-data-grid';
import { getEnvSelectionProcess } from '../../helpers/getEnvVaribles';
import { useAuthStore, useFetch } from '../../hooks';

const columns = [
  { field: 'id', headerName: 'ID', width: 130 },
  { field: 'company', headerName: 'Company', width: 130 },
  {
    field: 'score',
    headerName: 'Score',
    type: 'number',
    width: 80,
    editable: true,
  },
  {
    field: 'date',
    headerName: 'Date',
    width: 130,
    editable: true,
  },
  {
    field: 'time',
    headerName: 'Time',
    type: 'time',
    width: 80,
    editable: true,
  },
];

const selectionProcess = getEnvSelectionProcess();

export const AssesmentList = () => {
  const { id } = useAuthStore();
  const { data, loading } = useFetch(`${selectionProcess}/interviews/candidate/${id}`)

  console.log(loading, !data,  loading && !data)

  return (
    <Box>
      <Grid container justifyContent="center" mt={3} mb={3} >
        <Typography component="h1" variant="h4">
          List Interviews
        </Typography>
      </Grid>
      <DataGrid 
        columns={columns}
        rows={  
            data?.map((item, index) => ({
            id: index+1,
            company: item.company_id,
            score: item.score || 'Pending',
            date: item.date_interview.split('T')[0],
            time: item.date_interview.split('T')[1]
          })) || []
        }
        initialState={{
          pagination: {
            paginationModel: {
              pageSize: 5,
            },
          },
        }}
        pageSizeOptions={[5]}
        checkboxSelection
        disableRowSelectionOnClick
      />
    </Box >
  );
}