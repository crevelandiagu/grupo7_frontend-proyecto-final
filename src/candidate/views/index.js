export * from './CandidateDashboard';
export * from './NothingSelectedView';