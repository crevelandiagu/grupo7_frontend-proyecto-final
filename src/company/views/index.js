export * from './CompanyDashboard';
export * from './CreateEmployeeAccount';
export * from './CreateInterview';
export * from './Project';
export * from './SearchCandidate';
export * from './DetailProfile';
export * from './AssignProject';
export * from './Interview';
export * from './AssesmentList';
export * from './Assesment';
export * from './ContractList';
export * from './PerformanceList';
export * from './Performance';
