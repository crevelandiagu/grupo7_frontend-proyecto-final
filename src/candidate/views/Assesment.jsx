import Button from '@mui/material/Button';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import Container from '@mui/material/Container';

import { useAuthStore, useFetch } from '../../hooks';
import { Alert, FormControl, FormControlLabel, FormLabel, Radio, RadioGroup } from '@mui/material';
import { candidateApi, selectionProcess } from '../../api';


const sendAssesment = async (id = 1,) => {
  try {
    const { data } = await candidateApi.get(`/assement/take-exam/${id}/candidate`)
    console.log('data', data);
    return data.message;
  } catch (error) {
    console.log('error', error);
  }
}

const questions = [
  {
    question: "What is the capital of France?",
    options: ["Paris", "London", "Madrid", "Rome"]
  },
  {
    question: "Which planet is known as the Red Planet?",
    options: ["Mars", "Jupiter", "Venus", "Saturn"]
  },
  {
    question: "What is the largest ocean in the world?",
    options: ["Pacific Ocean", "Atlantic Ocean", "Indian Ocean", "Arctic Ocean"]
  },
  {
    question: "Who painted the Mona Lisa?",
    options: ["Leonardo da Vinci", "Pablo Picasso", "Vincent van Gogh", "Michelangelo"]
  },
  {
    question: "What is the chemical symbol for gold?",
    options: ["Au", "Ag", "Fe", "Cu"]
  }
];


export const Assesment = () => {

  const { id, errorMessage } = useAuthStore();
  const { data, loading } = useFetch(`${selectionProcess}/assement/take-exam/1/candidate`)

  const handleSubmit = (event) => {
    event.preventDefault();
    sendAssesment(id);
  };

  return (
    <Container component="main" maxWidth="xs">
      <Box
        sx={{
          marginTop: 8,
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center',
        }}
      >
        <Typography component="h1" variant="h4">
          Logic Test
        </Typography>
        <Box component="form" onSubmit={handleSubmit} sx={{ mt: 3 }}>
          {
            questions.map((question, index) => (
              <FormControl key={index} component="fieldset" sx={{ mb: 3 }}>
                <FormLabel component="legend">{question.question}</FormLabel>
                <RadioGroup
                  aria-label={`question-${index}`}
                  name={`question-${index}`}
                  defaultValue=""

                >
                  {question.options.map((option, optionIndex) => (
                    <FormControlLabel
                      key={optionIndex}
                      value={option}
                      control={<Radio size="small" />}
                      label={option}
                    />
                  ))}
                </RadioGroup>
              </FormControl>
            ))
          }
          <Grid item sx={{ mt: 2 }}
            xs={12}
            display={errorMessage ? '' : 'none'}
          >
            <Alert severity="error">{errorMessage}</Alert>
          </Grid>
          <Button
            type="submit"
            fullWidth
            variant="contained"
            sx={{ mt: 3, mb: 2 }}
          >
            Send
          </Button>
        </Box>
      </Box>
    </Container>
  );
}
