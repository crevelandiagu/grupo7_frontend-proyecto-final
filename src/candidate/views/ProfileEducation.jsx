import { useState } from 'react';

import Button from '@mui/material/Button';
import TextField from '@mui/material/TextField';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import Container from '@mui/material/Container';
import Alert from '@mui/material/Alert';
import Autocomplete from '@mui/material/Autocomplete';

import { useAuthStore, useCandidateStore, useForm } from '../../hooks';
import { candidateApi } from '../../api';
import { skillsListSelect } from '../../helpers/skillsListSelect';
import { useTranslation } from 'react-i18next';

const formData = {
  school: '',
  degree: '',
  startDate: '',
  endDate: '',
  location: '',
}

const saveEducationInfo = async (candidateId, { school: univerisity, endDate: end_date, degree: subject, skills, startDate: start_date }) => {
  try {
    const { data } = await candidateApi.post(`/profile/education/${candidateId}`, { univerisity, end_date, subject, skills, start_date })
    console.log('data', data);
    return data.message;
  } catch (error) {
    console.log('error', error);
  }
}

export const ProfileEducation = () => {

  const { id } = useAuthStore();
  const { t } = useTranslation();
  const { errorMessage } = useCandidateStore();

  const [formSubmitted, setFormSubmitted] = useState(false);
  const [skillsList, setSkillsList] = useState(null);

  const formValidations = {
    school: [(value) => value.length >= 3, t('message.inputSchool')],
    degree: [(value) => value.length >= 3, t('message.inputDegree')],
    startDate: [(value) => value.length >= 0, t('message.inputStartDate')],
    endDate: [(value) => value.length >= 0, t('message.inputEndDate')],
    location: [(value) => value.length >= 3, t('messsage.inputLocation')],
  }

  const {
    formState, school, degree, startDate, endDate, location, onInputChange, isFormValid,
    schoolValid, degreeValid, startDateValid, endDateValid, locationValid } = useForm(formData, formValidations);

  const handleSubmit = (event) => {
    event.preventDefault();
    setFormSubmitted(true);

    if (!isFormValid) return;
    saveEducationInfo(id, { ...formState, skills: skillsList });

  };

  return (
    <Container component="main" maxWidth="xs">
      <Box
        sx={{
          marginTop: 8,
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center',
        }}
      >
        <Typography component="h1" variant="h4">
          {t('education.title')}
        </Typography>
        <Box component="form" onSubmit={handleSubmit} sx={{ mt: 3 }}>
          <Grid container spacing={2}>
            <Grid item xs={12}>
              <TextField
                label={t('education.schoolTextFieldLabel')}
                type="text"
                placeholder={t('education.schoolTextFieldPlaceholder')}
                fullWidth
                name="school"
                value={school}
                onChange={onInputChange}
                error={!!schoolValid && formSubmitted}
                helperText={formSubmitted? schoolValid: ''}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                label={t('education.degreeTextFieldLabel')}
                type="text"
                placeholder={t('education.degreeTextFieldPlaceholder')}
                fullWidth
                name="degree"
                value={degree}
                onChange={onInputChange}
                error={!!degreeValid && formSubmitted}
                helperText={formSubmitted? degreeValid:''}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                label={t('education.startDateTextFieldLabel')}
                type="date"
                fullWidth
                name="startDate"
                value={startDate || "2023-01-01"}
                onChange={onInputChange}
                error={!!startDateValid && formSubmitted}
                helperText={formSubmitted? startDateValid: ''}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                label={t('education.endDateTextFieldLabel')}
                type="date"
                fullWidth
                name="endDate"
                value={endDate || "2023-01-01"}
                onChange={onInputChange}
                error={!!endDateValid && formSubmitted}
                helperText={formSubmitted?endDateValid: ''}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                label={t('education.locationTextFieldLabel')}
                type="text"
                placeholder={t('education.locationTextFieldPlaceholder')}
                fullWidth
                name="location"
                value={location}
                onChange={onInputChange}
                error={!!locationValid && formSubmitted}
                helperText={formSubmitted? locationValid: ''}
              />
            </Grid>
            <Grid item xs={12}>
              <Autocomplete
                id='skills'
                multiple
                limitTags={4}
                onChange={(event, value) => {
                  setSkillsList(value);
                }}
                options={skillsListSelect}
                getOptionLabel={(option) => option}
                // defaultValue={[skillList[1]]}
                filterSelectedOptions
                renderInput={(params) => (
                  <TextField
                    {...params}
                    label={t('education.skillTextFieldLabel')}
                    placeholder={t('education.skillTextFieldPlaceholder')}
                  />
                )}
              />
            </Grid>
          </Grid>
          <Grid item sx={{ mt: 2 }}
            xs={12}
            display={errorMessage ? '' : 'none'}
          >
            <Alert severity="error">{errorMessage}</Alert>
          </Grid>
          <Button
            type="submit"
            fullWidth
            variant="contained"
            sx={{ mt: 3, mb: 2 }}
          >
            {t('education.schoolTextFieldLabel')}
          </Button>
        </Box>
      </Box>
    </Container>
  );
}
