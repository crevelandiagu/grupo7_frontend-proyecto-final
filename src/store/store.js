import { configureStore } from '@reduxjs/toolkit'
import { authSlice } from './auth'
import { candidateSlice } from './candidate/candidateSlice'

export const store =  configureStore({
  reducer: {
    auth: authSlice.reducer,
    candidate: candidateSlice.reducer,
  },
})