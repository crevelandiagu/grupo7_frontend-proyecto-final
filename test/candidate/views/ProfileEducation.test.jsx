import { fireEvent, render } from '@testing-library/react';
import { ProfileEducation } from '../../../src/candidate/views/ProfileEducation';
import { store } from '../../../src/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

describe('Test to componenet <ProfileEducation />', () => {

  const valueSchool = 'Libre univerisity';
  const valueDegree = 'Engineer';
  const valueStartDate = '12-12-2016';
  const valueEndDate = '12-12-2021';
  const valueLocation = 'Bogota';

  test('Should be match with snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <ProfileEducation />
        </BrowserRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  })

  test('Call saveEducationInfo function', () => {

    const saveEducationInfo = jest.fn();

    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <ProfileEducation saveEducationInfo={saveEducationInfo} />
        </BrowserRouter>
      </Provider>
    );

    const inputSchool = container.querySelector('[name="school"]');
    const inputDegree = container.querySelector('[name="degree"]');
    const inputStartDate = container.querySelector('[name="startDate"]');
    const inputEndDate = container.querySelector('[name="endDate"]');
    const inputLocation = container.querySelector('[name="location"]');
    const inputSkill = container.querySelector('#skills');
    const form = container.querySelector('form');

    fireEvent.change(inputSchool, { target: { value: valueSchool } });
    fireEvent.change(inputDegree, { target: { value: valueDegree } });
    fireEvent.change(inputStartDate, { target: { value: valueStartDate } });
    fireEvent.change(inputEndDate, { target: { value: valueEndDate } });
    fireEvent.change(inputLocation, { target: { value: valueLocation } });
    fireEvent.change(inputSkill, { target: { value: { name: 'Docker' } } })
    fireEvent.submit(form)

    expect( saveEducationInfo ).toHaveBeenCalledTimes(0);
  })

});