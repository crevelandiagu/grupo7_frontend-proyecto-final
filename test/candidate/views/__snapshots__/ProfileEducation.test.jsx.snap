// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to componenet <ProfileEducation /> Should be match with snapshot 1`] = `
<div>
  <main
    class="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-binzgt"
    >
      <h1
        class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
      >
        education.title
      </h1>
      <form
        class="MuiBox-root css-178yklu"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r0:"
                id=":r0:-label"
              >
                education.schoolTextFieldLabel
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r0:"
                  name="school"
                  placeholder="education.schoolTextFieldPlaceholder"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      education.schoolTextFieldLabel
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r1:"
                id=":r1:-label"
              >
                education.degreeTextFieldLabel
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r1:"
                  name="degree"
                  placeholder="education.degreeTextFieldPlaceholder"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      education.degreeTextFieldLabel
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for=":r2:"
                id=":r2:-label"
              >
                education.startDateTextFieldLabel
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r2:"
                  name="startDate"
                  type="date"
                  value="2023-01-01"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-14lo706"
                  >
                    <span>
                      education.startDateTextFieldLabel
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for=":r3:"
                id=":r3:-label"
              >
                education.endDateTextFieldLabel
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r3:"
                  name="endDate"
                  type="date"
                  value="2023-01-01"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-14lo706"
                  >
                    <span>
                      education.endDateTextFieldLabel
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r4:"
                id=":r4:-label"
              >
                education.locationTextFieldLabel
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r4:"
                  name="location"
                  placeholder="education.locationTextFieldPlaceholder"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      education.locationTextFieldLabel
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1h51icj-MuiAutocomplete-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="skills"
                  id="skills-label"
                >
                  education.skillTextFieldLabel
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-invalid="false"
                    autocapitalize="none"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    id="skills"
                    placeholder="education.skillTextFieldPlaceholder"
                    role="combobox"
                    spellcheck="false"
                    type="text"
                    value=""
                  />
                  <div
                    class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                  >
                    <button
                      aria-label="Open"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                      tabindex="-1"
                      title="Open"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="ArrowDropDownIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        education.skillTextFieldLabel
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1vddya1-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-1hdyuqk-MuiPaper-root-MuiAlert-root"
            role="alert"
          >
            <div
              class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="ErrorOutlineIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
            />
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          education.schoolTextFieldLabel
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </main>
</div>
`;
